src/test/Assets/TestProjects/StartupHookWithDependency/StartupHookWithDependency.cs

```diff
@@ -0,0 +1,12 @@
+using System;
+
+internal class StartupHook
+{
+    public static void Initialize()
+    {
+        Console.WriteLine("Hello from startup hook with dependency!");
+
+        // A small operation involving NewtonSoft.Json to ensure the assembly is loaded properly
+        var t = typeof(Newtonsoft.Json.JsonReader);
```

Comment:
A few additional tests we should add is around creating a CustomerAssemblyLoad context and one that preloads assemblies (dynamically).  This is one of the core golden path scenarios and we should ensure it is functional.

Reply:
As discussed, the preloading scenario won't work due to bootstrapping issues. I don't think there's a good testcase for a custom ALC here either - that seems like it belongs to the plugin model work.